projects/laji/src/app/shared-modules/select/select/select.component.ts

```diff
@@ -34,21 +42,24 @@ export class SelectComponent implements OnInit, OnChanges, OnDestroy {
   @Input() title: string;
   @Input() filterPlaceHolder = 'Search...';
   @Input() useFilter = true;
-  @Input() selected: string[] = [];
+  @Input() selected: (string|SelectedOptions)[] = [];
   @Input() open = false;
   @Input() disabled = false;
   @Input() outputOnlyId = false;
-  @Output() selectedChange = new EventEmitter<string[]|string>();
+  @Output() selectedChange = new EventEmitter<(SelectedOptions|string)[]|string>();
   @Input() multiple = true;
   @Input() info: string;
   @Input() loading = false;
+  @Input() checkboxType = CheckboxType.basic;
+  @Input() class: string[] = ['subCategoriesOptions', 'subCategoriesContainer', 'subMenu'];
   @ViewChild('filter') filter: ElementRef;
 
-  selectedOptions: SelectOptions[] = [];
-  unselectedOptions: SelectOptions[] = [];
+  selectedOptions: (SelectedOptions|SelectOptions)[] = [];
+  unselectedOptions: (SelectedOptions|SelectOptions)[] = [];
```

Comment:
Now in select component "SelectedOptions" are not used. I leave still that point open because i need to clean the code and to check more because it seems it's working well, but sometimes i find some bugs. Tomorrow morning i will check more and later i will resolve that point